app/colorCoder.js

```diff
@@ -1,99 +1,41 @@
-}
-
-function getColorFromPairNumber(pairNumber)
-{
-	let minorSize = MajorColorNames.length;
-	let majorSize = MinorColorNames.length;
-	
-	if (pairNumber < 1 || pairNumber > minorSize * majorSize)
- 	{
- 		throw `Argument PairNumber:${pairNumber} is outside the allowed range` 
-	}
-	let zeroBasedPairNumber = pairNumber - 1;
-    let majorIndex = parseInt (zeroBasedPairNumber / minorSize);
-    let minorIndex = parseInt(zeroBasedPairNumber % minorSize);
-	let  pair = new ColorPair();
-	pair.majorColor = MajorColorNames[majorIndex];
-	pair.minorColor = MinorColorNames[minorIndex];
-	return pair;
-}
-
-function getPairNumberFromColor(pair)
-        {
-        let majorIndex = -1;
-        for (let i = 0; i < MajorColorNames.length; i++)
-            {
-                if (MajorColorNames[i] == pair.majorColor)
-                {
-                    majorIndex = i;
-                    break;
-                }
-            }
-
-        let minorIndex = -1;
-            for (let i = 0; i < MinorColorNames.length; i++)
-            {
-                if (MinorColorNames[i] == pair.minorColor)
-                {
-                    minorIndex = i;
-                    break;
-                }
-            }
-    
-        if (majorIndex == -1 || minorIndex == -1)
-            {
-                throw `Unknown Colors:${pair.toString()}`;
-            }
-
-        return (majorIndex * MinorColorNames.length) + (minorIndex + 1);
-	 }
-	 
-	 function test(){
-			pairNumber = 4;
-            let testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "WHITE");
-			console.assert(testPair1.minorColor == "BROWN");
-
-			pairNumber = 5;
-            testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "WHITE");
-			console.assert(testPair1.minorColor == "SLATEGRAY");
-			 
-			pairNumber = 23;
-            testPair1 = getColorFromPairNumber(pairNumber);
-            console.log(`[In]Pair Number: ${pairNumber},[Out] Colors:${testPair1}`);
-			console.assert(testPair1.majorColor == "RED");
-			console.assert(testPair1.minorColor == "GREEN");
-			 
-            let testPair2 = new ColorPair();
-			testPair2.majorColor="YELLOW";
-			testPair2. minorColor ="GREEN";
-            pairNumber =getPairNumberFromColor(testPair2);
-            console.log(`[In]Colors: ${testPair2}, [Out] PairNumber: ${pairNumber}`);
-			console.assert(pairNumber==18);
-
-			testPair2 = new ColorPair();
-			testPair2.majorColor="RED";
-			testPair2. minorColor ="BLUE";
-            pairNumber =getPairNumberFromColor(testPair2);
-            console.log(`[In]Colors: ${testPair2}, [Out] PairNumber: ${pairNumber}`);
-			console.assert(pairNumber==6);
+import { getColorFromPairNumber, getPairNumberFromColor } from './colorPairFunctions.js';
```

Comment:
Naming of this file can be better, to indicate the content (this file contains only test code, hence something like testColorCode.js ! )